gcc/rust/checks/errors/privacy/rust-privacy-reporter.cc

```diff
@@ -34,33 +35,23 @@ PrivacyReporter::PrivacyReporter (
     current_module (tl::nullopt)
 {}
 
-// Find a proc_macro, proc_macro_derive or proc_macro_attribute
-// attribute in a vector of attribute
-static tl::optional<std::string>
-find_proc_macro_attribute (const AST::AttrVec &outer_attrs)
-{
-  for (auto &a : outer_attrs)
-    {
-      auto &segments = a.get_path ().get_segments ();
-      if (segments.size () != 1)
-	continue;
-      auto name = segments.at (0).get_segment_name ();
-      if (name == Values::Attributes::PROC_MACRO
-	  || name == Values::Attributes::PROC_MACRO_ATTRIBUTE
-	  || name == Values::Attributes::PROC_MACRO_DERIVE)
-	return name;
-    }
-
-  return tl::nullopt;
-}
-
 // Common check on crate items when dealing with 'proc-macro' crate type.
 static void
 proc_macro_privacy_check (std::unique_ptr<HIR::Item> &item)
 {
   if (item->get_hir_kind () == HIR::Node::BaseKind::VIS_ITEM)
     {
-      auto attribute = find_proc_macro_attribute (item->get_outer_attrs ());
+      tl::optional<std::string> attribute = tl::nullopt;
+      // Find a proc_macro, proc_macro_derive or proc_macro_attribute
+      // attribute in a vector of attribute
+      for (const auto &attr : item->get_outer_attrs ())
+	{
+	  if (Analysis::Attributes::is_proc_macro_type (attr) != tl::nullopt)
```

Comment:
this seems odd i think you should be storing the result here in a variable instead of calling it twice like this